core/state/trie_prefetcher.go

```diff
@@ -159,32 +122,19 @@ func (p *triePrefetcher) prefetch(owner common.Hash, root common.Hash, addr comm
 // trie returns the trie matching the root hash, or nil if the prefetcher doesn't
 // have it.
 func (p *triePrefetcher) trie(owner common.Hash, root common.Hash) Trie {
```

Comment:
Once the trie is retrieved, the corresponding subPrefetcher is terminated.

It's problematic for the blocks before ByzantiumFork as the trie is retrieved after each transaction, instead of at the block boundary.

Reply:
I feel like we should make it as a block function. If there are some ongoing prefetching tasks, the function will be blocked until these tasks are fully executed.

Reply:
Done in the latest commit.